blockly-ws-search/workspace_search.js

```diff
@@ -191,6 +192,22 @@ class WorkspaceSearch {
     }
   }
 
+  /**
+   * Handles clicking on the input value in search bar.
+   * @param {Event} e The onclick event.
+   */
+  onInputClick_() {
+    const oldBlock = this.currentBlock_;
+    let oldIdx = this.currentBlockIndex_;
+    this.search();
+    const currentBlockIdx = this.blocks_.indexOf(oldBlock);
+    if (currentBlockIdx > -1) {
+      this.setCurrentBlock_(currentBlockIdx);
+    } else {
+      this.setCurrentBlock_(oldIdx);
```

Comment:
Search will already set a new current block. If we want to change which one is selected, then I think it should only happen if the old block exists in new search results (i.e., remove else).

Additionally, this is behavior I think should go inside of `search` (so it always happens), possibly with a flag to allow turning off for performance optimization. That way if for example you are one a block that matches and you add/remove letters to search text, and that same "current" block is still in search pool it will remain highlighted. Also by moving it into `search` we can avoid calling `setCurrentBlock_` twice in one search unnecessarily.

Reply:
I don't really agree with the first part. I think it is much nicer functionality to start searching where we left off. I think for a small ws it doesn't really matter but if someone has a lot of blocks it would be pretty jarring to be editing code in one part and then be taken to the start of the workspace when we click in the box. 

However, I updated it to be the functionality you described for this part and will take a look at adding extra functionality in another PR.